src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -2,6 +2,36 @@
 
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        System.out.println();
+        String[] strDateFrom = dateFrom.split("\\.");
+        int numDateFrom = Integer.parseInt(strDateFrom[2]) * 10000
+                + Integer.parseInt(strDateFrom[1]) * 100
+                + Integer.parseInt(strDateFrom[0]);
+        String[] strDateTo = dateTo.split("\\.");
+        String[] returnStrings = new String[names.length];
+        int numDateTo = Integer.parseInt(strDateTo[2]) * 10000
+                + Integer.parseInt(strDateTo[1]) * 100
+                + Integer.parseInt(strDateTo[0]);
+        for (int i = 0; i < names.length; i++) {
+            int sum = 0;
+            for (String datum : data) {
+                String[] array = datum.split(" ");
+                String[] strCurrentDate = array[0].split("\\.");
+                int numCurrentDate = Integer.parseInt(strCurrentDate[2]) * 10000
+                        + Integer.parseInt(strCurrentDate[1]) * 100
+                        + Integer.parseInt(strCurrentDate[0]);
+                if (names[i].equals(array[1]) && numCurrentDate >= numDateFrom
+                        && numCurrentDate <= numDateTo) {
+                    sum += Integer.parseInt(array[2]) * Integer.parseInt(array[3]);
+                }
+            }
+            returnStrings[i] = names[i] + " - " + sum;
+        }
+        StringBuilder returnString = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        for (String string : returnStrings) {
```

Comment:
[Don’t create redundant loops. Two is enough. ](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#dont-create-redundant-loops-two-is-enough)